src/libs/ReportUtils.js

```diff
@@ -883,11 +883,11 @@ function buildOptimisticIOUReport(ownerEmail, userEmail, total, chatReportID, cu
  */
 function getIOUReportActionMessage(type, total, participants, comment, currency, paymentType = '', isSettlingUp = false) {
     const amount = NumberFormatUtils.format(preferredLocale, total / 100, {style: 'currency', currency});
-    const displayNames = _.map(participants, participant => getDisplayNameForParticipant(participant.login, true));
+    const displayNames = _.map(participants, participant => Str.htmlEncode(getDisplayNameForParticipant(participant.login, true)));
     const who = displayNames.length < 3
         ? displayNames.join(' and ')
         : `${displayNames.slice(0, -1).join(', ')}, and ${_.last(displayNames)}`;
-
+    const encodedComment = Str.htmlEncode(comment);
```

Comment:
Why are we encoding everything again? Is it because we decode [here](https://github.com/Expensify/App/pull/15578/files#diff-14a3e24befcdfefed9da2b90241ce631d06501ceff6144bce830652c899cdccfR9)?

Reply:
The comments are being decoded here https://github.com/Expensify/App/blob/a7c196fb09e06ed31dfe2f520da134515c835386/src/components/ReportActionItem/IOUQuote.js#L86 So I encoded the comments to keep them well displayed (the logic is similar to what we did with the display name). Otherwise, before we received an encoded comment in response from BE, this issue will happen 

https://user-images.githubusercontent.com/113963320/222876440-10df35f6-90b3-4379-91b1-4864d75351d2.mp4

If we should not care the message, I wil remove the encode.

Reply:
What are your thoughts @pecanoro?

Reply:
Oh boy, yeah, it's not your fault, as I mentioned, the way we encode and decode messages got a bit chaotic. Anyways, I will finish testing this and I will be ready to merge if I can't find any issues.

Reply:
Last question, did you make sure XSS injections are not possible with the new changes? I am going to test it, but two people testing it is even better.

Reply:
I think React does it for us. You can check the reference here: https://reactjs.org/docs/introducing-jsx.html#jsx-prevents-injection-attacks. I also did some tests for this problem, and everything was ok. Thanks

Reply:
Can you retest this after merging with main? It is not working for me and I think it's due to main changing:

![image](https://user-images.githubusercontent.com/6474442/223200302-0705953c-e51f-4306-be30-4d5340f1838a.png)

Reply:
https://user-images.githubusercontent.com/113963320/223317343-515bad77-9d06-41af-a87b-b93c8ed027b2.mp4
I tested and it's not working now. I think the problem is in BE, they don't encode the comment (they used to do that). Hence we receive this message in the response from BE 
<img width="805" alt="image" src="https://user-images.githubusercontent.com/113963320/223318238-67c9f484-e6bc-48fb-8a60-f50a4aaa5de1.png">
Moreover, we are decoding "text" to get the display message. That is why the message "&nbsp" is decoded and we can only see a blank. https://github.com/Expensify/App/blob/6413a798a64f179541eee79778102563d0c0a74d/src/components/ReportActionItem/IOUQuote.js#L86
@pecanoro, do you have any suggestion?

Reply:
I think it might have stopped working after adding this https://github.com/Expensify/App/pull/15578. Does it work if we revert that and we use your solution?

Reply:
I don't think so. Because I'm fixing the issue with IOU message in report actions list 
<img width="1920" alt="Screen Shot 2023-03-07 at 23 45 02" src="https://user-images.githubusercontent.com/113963320/223491011-9efe7687-fcbf-4c2c-afe9-dad13bdfa31d.png">

https://github.com/Expensify/App/pull/15578 is fixing the issue with IOU message in the details
<img width="1920" alt="Screen Shot 2023-03-07 at 23 47 04" src="https://user-images.githubusercontent.com/113963320/223491215-cb0c4bc6-954f-4717-b88e-fb3d60a41a7c.png">
As I mentioned, the problem is from BE, they don't encode the comment (they used to do that).

Reply:
I suggest we apply a similar solution to what the author of https://github.com/Expensify/App/pull/15578 did https://github.com/Expensify/App/pull/15578/files
Instead of decoding the text form of message, we can decode the html form because it is well encoded before send from BE.
The change will be like this:
```diff
diff --git a/src/components/ReportActionItem/IOUQuote.js b/src/components/ReportActionItem/IOUQuote.js
index 032297505a..17d29b3597 100644
--- a/src/components/ReportActionItem/IOUQuote.js
+++ b/src/components/ReportActionItem/IOUQuote.js
@@ -83,7 +83,7 @@ const IOUQuote = props => (
                         : styles.cursorDefault]}
                     >
                         {/* Get remainder of IOU message */}
-                        {Str.htmlDecode(fragment.text.substring(fragment.text.indexOf(' ')))}
+                        {Str.htmlDecode(fragment.html.substring(fragment.html.indexOf(' ')))}
                     </Text>
                 </Text>
                 <Icon src={Expensicons.ArrowRight} fill={props.shouldAllowViewDetails ? StyleUtils.getIconFillColor(getButtonState(props.isHovered)) : themeColors.transparent} />
```
It works after the change applied

https://user-images.githubusercontent.com/113963320/223495846-f0a936b2-5719-4eb2-8f56-e00043b747d7.mp4

Reply:
I have pushed the suggested change. You can test the IOU request to verify it is working now.

Reply:
Yes, I think we need to use html here as well, not sure why we would use text

Reply:
Are we good to continue with the change? I expect to have the PR merged soon.

Reply:
Yes, I am testing everything again